File: 백준/Silver/9012. 괄호/괄호.py
import sys

input = sys.stdin.readline

inputs = [input().strip() for _ in range(int(input()))]

def isCorrect(li):
    stk = []
    for i in li:
        if i == ")":
            if not stk or stk[-1] != "(":
                return False
            
            stk.pop()
            continue
            
        stk.append(i)
        
    return not stk
        

for i in inputs:
    li = list(i)
    print("YES" if isCorrect(li) else "NO")

File: 백준/Silver/9012. 괄호/괄호.js
const fs = require('fs')

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split("\n").slice(1)

function isCorrect(arr) {
  const stk = []
  for(let s of arr){
    if(s === ")"){
      if(stk.at(-1) !== "(")return false
      stk.pop()
      continue
    }
    stk.push(s)
  }

  return stk.length === 0
}

for(let s of inputs){
  console.log(isCorrect(s.split("")) ? "YES" : "NO")
}